src/ReturnTypes/EloquentBuilderExtension.php

```diff
@@ -98,9 +98,13 @@ public function getTypeFromMethodCall(
             );
         }
 
-        // 'get' method return type
-        if ($methodReflection->getName() === 'get') {
-            return new GenericObjectType(Collection::class, [$modelType]);
+        // collection return methods...
+        if (in_array($methodReflection->getName(), BuilderHelper::COLLECTION_RETRIEVAL_METHODS, true)) {
+            $class = $modelType->getReferencedClasses()[0];
+
+            $collection = (new $class)->newCollection([]);
+
+            return new GenericObjectType(get_class($collection), [$modelType]);
```

Comment:
We can do this in a more general way. Similar to how custom Eloquent builder is implemented. We can extract the logic of getting the Collection class to a method. Like [here.](https://github.com/nunomaduro/larastan/blob/master/src/Methods/BuilderHelper.php#L133-L148) It can be also in the `BuilderHelper` class I guess.

Then we can have another `if` here and that would check if `Eloquent\Collection` is referenced in return type. With something like `in_array(Collection::class, $returnType->getReferencedClasses(), true)` 
Then we can just do `$returnType = new GenericObjectType($collectionClass, [$modelType])`